src/bytes.rs

```diff
@@ -979,6 +1035,30 @@ unsafe fn shared_clone(data: &AtomicPtr<()>, ptr: *const u8, len: usize) -> Byte
     shallow_clone_arc(shared as _, ptr, len)
 }
 
+unsafe fn shared_to_vec_impl(shared: *mut Shared, ptr: *const u8, len: usize) -> Vec<u8> {
+    // This fence is needed for the same reason in release_shared.
+    if (*shared).ref_cnt.load(Ordering::Acquire) == 1 {
+        let buf = (*shared).buf;
+        let cap = (*shared).cap;
+
+        // Deallocate Shared
+        drop(Box::from_raw(shared as *mut mem::ManuallyDrop<Shared>));
+
+        // Copy back buffer
+        ptr::copy(ptr, buf, len);
+
+        Vec::from_raw_parts(buf, len, cap)
+    } else {
+        let v = slice::from_raw_parts(ptr, len).to_vec();
+        release_shared(shared);
+        v
+    }
+}
```

Comment:
Since this is essentially an [`Arc::try_unwrap`](https://doc.rust-lang.org/stable/std/sync/struct.Arc.html#method.try_unwrap), please use the same logic as the standard library does for updating the reference-count.

Reply:
Done.

It now uses `compare_exchange` to set the `ref_cnt` to `0` using `AcqRel` if the `Shared` is unique.
Otherwise, it is equivalent to a `Relaxed` load.